contracts/contracts/rem.attr/include/rem.attr/rem.attr.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <eosio/eosio.hpp>
+
+
+namespace eosio {
+
+   class [[eosio::contract("rem.attr")]] attribute : public contract {
+   public:
+      using contract::contract;
+
+      [[eosio::action]]
+      void confirm( const name& owner, const name& attribute_name );
+
+      [[eosio::action]]
+      void create( const name& attribute_name, int32_t type, int32_t ptype );
+
+      [[eosio::action]]
+      void setattr( const name& issuer, const name& receiver, const name& attribute_name, const std::vector<char>& value );
+
+      [[eosio::action]]
+      void unsetattr( const name& issuer, const name& receiver, const name& attribute_name );
+
+      using confirm_action   = eosio::action_wrapper<"confirm"_n,     &attribute::confirm>;
+      using create_action    = eosio::action_wrapper<"create"_n,       &attribute::create>;
+      using setattr_action   = eosio::action_wrapper<"setattr"_n,     &attribute::setattr>;
+      using unsetattr_action = eosio::action_wrapper<"unsetattr"_n, &attribute::unsetattr>;
+
+   private:
+      enum class data_type : int32_t { Boolean = 0, Int, LargeInt, ChainAccount, UTFString, DateTimeUTC, CID, OID, Binary, Set, MaxVal };
+      enum class privacy_type : int32_t { SelfAssigned = 0, PublicPointer, PublicConfirmedPointer, PrivatePointer, PrivateConfirmedPointer, MaxVal };
+
+      struct [[eosio::table]] attribute_info {
+         name    attribute_name;
+         int32_t type;
+         int32_t ptype;
+
+         uint64_t primary_key() const { return attribute_name.value; }
+      };
+
+      struct [[eosio::table]] attribute_data {
+         name               attribute_name;
+         name               issuer;
+         std::vector<char>  data;
```

Comment:
@me-vadim could you please remind me why we decide to store data in bytes?

Reply:
@mark-tsyrulnyk because fc::variant can`t be used in contracts (without altering eosio.cdt)